### ISSUE TITLE
[Discussion] ORT GPU binaries do not contain DML

### ISSUE THREAD

**Issue by gedoensmax** [microsoft/onnxruntime#20638]
### Describe the issue

From what I understand DirectML is considered the default GPU backend on Windows systems. Nonetheless the "GPU" [build](https://onnxruntime.ai/docs/get-started/with-cpp.html#builds) does not contain DML. In my opinion it would make more sense to have the GPU package include DirectML besides CUDA and TRT.  This package will then also run on any GPU. In addition such package can dynamically load OpenVino for additional GPU support on Intel.

### To reproduce

When downloading a release package just call [GetAvailableProviders()](https://onnxruntime.ai/docs/api/c/namespace_ort.html#ac2f785275bced9e6853a2f4ea6b71c04). This will report the supported providers. Another problem is that when dynamically loading OpenVINO it will never report it as supported provider.

### Platform

Windows

### OS Version

11

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.17.3

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

CUDA, DirectML, TensorRT

**Comment by gedoensmax** [comment]
@snnn or @pranavsharma can you help with this ?

**Comment by tenten8401** [comment]
+1 on this, currently making a custom build for DirectML support

**Comment by saidwho12** [comment]
I agree with this, making a combined build would make things so much easier for me, in my software I want to dynamically switch between DirectML and TRT without needing to have 2 different EXEs
